File: src/typescript/transaction-builder/hive-apps-operations/follow-operation.ts
import { createHiveChain, FollowOperation } from '@hiveio/wax';

// Create chain
const chain = await createHiveChain();

// Create transaction with data from remote
const tx = await chain.createTransaction();

const { wallet, publicKey1 } = globalThis.snippetsBeekeeperData; // It should be the public key of the account that you authorize the operation

// Your account name
const yourAccount = 'your_account';

// Blog author name to follow
const blogToFollow = 'interesting_blog';

// Blog author name to mute
const blogToMute = 'spammer';

// Author of post to reblog
const toReblog = 'reblog_me';

// Create follow operation new instance
const followOperation = new FollowOperation();

// Push operations of follow operation into the created transaction
tx.pushOperation(
  followOperation
    .followBlog(yourAccount, blogToFollow)
    .muteBlog(yourAccount, blogToMute)
    .reblog(yourAccount, toReblog, 'post_permlink')
    // The account that authorizes underlying custom json operation is also reponsible for signing the transaction using its posting authority
    .authorize(yourAccount)
);

// Sign and build the transaction
tx.sign(wallet, publicKey1);

console.log(tx.transaction.operations[0]); // Follow operation
console.log(tx.transaction.operations[1]); // Mute operation
console.log(tx.transaction.operations[2]); // Reblog operation

File: src/rust/transaction/hive_apps_operations/follow_operation.rs
use runner::{SnippetsBeekeeperData, snippets_beekeeper_data};
use wax::prelude::*;
use wax::hive_apps_operations::HiveAppsOperation;
use wax::hive_apps_operations::FollowOperation;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    // Create chain
    let chain = create_hive_chain(None)?;

    // Create transaction with data from remote
    let mut tx = chain.create_transaction(None).await?;

    let SnippetsBeekeeperData { signer1, public_key1, .. } = snippets_beekeeper_data();

    // Your account name
    let your_account = "your-account";

    // Blog author name to follow
    let blog_to_follow = "interesting-blog";

    // Blog author name to mute
    let blog_to_mute = "spammer";

    // Author of post to reblog
    let to_reblog = "reblog-me";

    // Create follow operation new instance
    let follow_operation = FollowOperation::new()
        .follow_blog(your_account, vec![blog_to_follow.into()])?
        .mute_blog(your_account, vec![blog_to_mute.into()])?
        .reblog(your_account, to_reblog, "post-permlink")
        // The account that authorizes underlying custom json operation is
        // also responsible for signing the transaction using its posting authority
        .authorize(vec![your_account.into()], Vec::new())?;

    // Push operations of follow operation into the created transaction
    tx.push_complex_operation(&chain, follow_operation)?;

    // Sign and build the transaction
    tx.sign(&signer1, &public_key1)?;

    println!("{:#?}", tx.transaction().operations[0]); // Follow operation
    println!("{:#?}", tx.transaction().operations[1]); // Mute operation
    println!("{:#?}", tx.transaction().operations[2]); // Reblog operation

    Ok(())
}
